src/Shared/CommunicationsUtilities.cs

```diff
@@ -490,86 +493,85 @@ internal static async Task<int> ReadAsync(Stream stream, byte[] buffer, int byte
         /// <summary>
         /// Given the appropriate information, return the equivalent TaskHostContext.  
         /// </summary>
-        internal static TaskHostContext GetTaskHostContext(IDictionary<string, string> taskHostParameters)
+        internal static HandshakeOptions GetTaskHostContext(IDictionary<string, string> taskHostParameters)
```

Comment:
Rename to GetHandshakeOptions to keep in line with the new enum name?

Reply:
Actually, wouldn't it be cleaner if you moved all the GetTaskHostContext methods as static constructors on the enum instead, and rename to `From`: `HandshakeOptions.From(bool, bool)`

---
In reply to: [412594545](https://github.com/microsoft/msbuild/pull/5196#discussion_r412594545) [](ancestors = 412594545)

Reply:
Can an enum have a method like a class? I think I'd have to replace the enum with a class, and that would make it much bigger.

Reply:
You can write extension methods for it: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-create-a-new-method-for-an-enumeration

Reply:
I put a change [here](https://github.com/Forgind/msbuild/commit/001c9b69a6e5b8bcf68355f753b5f2234c11fa6b), but I don't think I like it. It would probably be good if there were a lot of classes that were going to touch `HandshakeOptions`, but since all references to it are focused in this class, it seems unnecessary.